File: Chapitre3/exercice3.3.py
import os
import sys

def main():
    fichier = input("Entrer le nom du repertoire: ")
    try:
        os.execl("/usr/bin/vim", "vim", fichier)
    except OSError as e:
        print(f"Erreur execv: {e}", file=sys.stderr)
        sys.exit(1)

if __name__ == "__main__":
    main()

File: Chapitre3/exercice3.3.c
//saisis du nom de fichier ne texte et ouvre ce fichier dans l'éditeur emacs dont le fichier se trouve à l'emplacement /usr/bin/emacs
#include <stdlib.h>
#include <stdio.h>
#include <string.h>
#include <unistd.h>

int main() {
    char fichier[255];
    printf("Entrer le nom du repertoire:");
    scanf("%s", fichier);
    execl("/usr/bin/vim", "vim", fichier , NULL);
    
    perror("Erreur execv");
    return 1;
}